### ISSUE TITLE
Question: Why do we compare version of xz in compile time with version in runtime in tests?

### ISSUE THREAD

**Issue by michal-josef-spacek** [pmqs/Compress-Raw-Lzma#13]
We have t/001version.t test file and comparison between LZMA_VERSION and Compress::Raw::Lzma::lzma_version_number etc.
What is the reason for this comparison?

context:
In Fedora (in case of perl-Compress-Raw-Lzma package) we discussed this in https://src.fedoraproject.org/rpms/perl-Compress-Raw-Lzma/pull-request/3.

**Comment by pmqs** [comment]
It is there to guard against you building with version "x" of a library, but then running the module with version "y" of the library, which has a different API.

**Comment by michal-josef-spacek** [comment]
I understand this, but it's breaking in the case of moving from xz 5.6.2 to 5.6.3 in e.g. Fedora. And this is not a fine intention. The API should be the same.

**Comment by pmqs** [comment]
Should Fedora not just update this module at the same time then? 

Alternatively, just set the `TEST_SKIP_VERSION_CHECK` environment variable to get the test skipped. Only caveat if you do use `TEST_SKIP_VERSION_CHECK`  is you need to be sure that the API hasn't changed.

In the past, before I added this check to my modules that interface to C libraries, I would get error reports where someone was attempting to build with version "x" of the header file, but version "y" of the library. In your use-case it probably wouldn't matter, but I've had plenty where the code crashed because the API was different.

**Comment by michal-josef-spacek** [comment]
Workflow in Fedora:
1) All packages are built using the minimal system with actual versions. So the problem in infrastructure build with the version isn't a issue.
2) Since this test exists, the package author probably added a strict dependency on the xz package. This is not fine.

ad 2) 
We removed the strict dependency to xz, this is ok.
We added TEST_SKIP_VERSION_CHECK to skip actually. Probably not needed. I am thinking if we need or not.

FYI: We are packaging upstream tests to infrastructure testing. So in the test package I need to skip this test, because a situation could be a) build with xz 5.6.2, b) xz increase to 5.6.3, without rebuild of Compress::Raw::Lzma c) test via upstream test.
